eth/downloader/peer.go

```diff
@@ -131,6 +167,36 @@ func (p *peer) SetIdle() {
 	atomic.StoreInt32(&p.idle, 0)
 }
 
+// SetIdle sets the peer to idle, allowing it to execute new retrieval requests.
```

Comment:
Yes it is needed. A lot of peers in the network are very slow and if we retrieve a big batch of blocks they will only time out and block up the entire sync algo. The point of the throttling mechanism here is that even though we pull blocks concurrently from many peers, we can actually measure the speed of each and pull only as much as they can provide in a reasonable time without having to wait for them.